node/src/components/block_accumulator/block_acceptor.rs

```diff
@@ -114,13 +120,17 @@ impl BlockAcceptor {
         &mut self,
         finality_signature: FinalitySignature,
         peer: Option<NodeId>,
+        validator_slots: u32,
     ) -> Result<Option<FinalitySignature>, AcceptorError> {
         if self.block_hash != finality_signature.block_hash {
             return Err(AcceptorError::BlockHashMismatch {
                 expected: self.block_hash,
                 actual: finality_signature.block_hash,
             });
         }
+        if let Some(node_id) = peer {
+            check_signatures_from_peer_bound(validator_slots * 2, node_id, &self.signatures)?;
```

Comment:
It would be a good idea to leave a comment explaining why we multiply by 2 here.

Reply:
Done. I copy/pasted the comment from Rafał above.